numpy/lib/tests/test_function_base.py

```diff
@@ -1234,6 +1234,85 @@ def test_empty(self):
         assert_array_equal(b, np.array([0, 1]))
 
 
+class TestHistogramOptimBinNums(TestCase):
+    """
+    Provide test coverage when using provided estimators for optimal number of bins
+    """
+
+    def test_empty(self):
+        estimator_list = ['fd', 'scott', 'rice', 'sturges', 'auto']
+        # check it can deal with empty data
+        for estimator in estimator_list:
+            a, b = histogram([], bins=estimator)
+            assert_array_equal(a, np.array([0]))
+            assert_array_equal(b, np.array([0, 1]))
+
+    def test_incorrect_methods(self):
+        """
+        Check a Value Error is thrown when an unknown string is passed in
+        """
+        check_list = ['mad', 'freeman', 'histograms', 'IQR']
+        for estimator in check_list:
+            try:
+                a, b = histogram([1, 2, 3], estimator)
+                # note, no method runs if the dataset is empty
+            except ValueError:
+                # This is expected
+                pass
+            else:
+                raise AssertionError("np.histogram ran when passed "
+                                     "unknown automatic method: {}".format(estimator))
```

Comment:
Check out `assert_raises` -- I think you'll find it awesome :-).

(Also will fix the minor issue of plain `{}` being illegal in py26 format strings.)